feat: honour `maxTasksPerRound` from spark-api

bajtos opened the pull request:
Links:
- https://github.com/filecoin-station/spark-api/pull/172
- https://github.com/filecoin-station/roadmap/issues/52

bajtos commented on lib/constants.js:
```diff
@@ -1,3 +1,3 @@
 export const SPARK_VERSION = '1.6.0'
-export const DELAY_BETWEEN_RETRIEVALS = 10_000
 export const MAX_CAR_SIZE = 200 * 1024 * 1024 // 200 MB
+export const APPROX_ROUND_LENGTH_IN_MS = 60 * 60_000 // 1 hour
```
Let's keep this initial version simple.

In the future, we can query Meridian state to find out how many blocks remain in the current round and calculate the remaining round time using that value.

juliangruber commented on lib/spark.js:
```diff
@@ -1,14 +1,15 @@
 export default class Spark {
   #fetch
   #activity = new ActivityState()
+  #maxTasksPerNode = 360
```
```suggestion
  #maxTasksPerNode
```

instead of assigning a default that could be vastly off, what about not using the variable until we have fetched a value?

bajtos replied:
What if the first iteration fails to fetch the value (e.g. because we are offline)? We need to know the delay before we start the second iteration. See https://github.com/filecoin-station/spark/pull/47/files#diff-9eb2d17c6352497045e4a22df4b4957d5d44f3cfff0c66c6519d07705ee1ed86R177

juliangruber commented on lib/spark.js:
```diff
@@ -169,7 +173,12 @@ export default class Spark {
       }
-      await sleep(DELAY_BETWEEN_RETRIEVALS)
+      const duration = Date.now() - started
+      const baseDelay = APPROX_ROUND_LENGTH_IN_MS / this.#maxTasksPerNode
```
what about instead having `this.#delayBetweenRetrievals = APPROX_ROUND_LENGTH_IN_MS / maxTasksPerNode` and storing that in `getRetrieval()`?

And then something like

```js
await sleep(Math.max(0, this.#delayBetweenRetrievals - duration))
```

bajtos replied:
See https://github.com/filecoin-station/spark-api/pull/172#discussion_r1423969809

I designed the current solution with a longer-term vision in my mind.

If you feel strongly this is YAGNI, then I can rework this PR to apply your suggestion. Let me know!

juliangruber commented on lib/spark.js:
```diff
@@ -23,9 +24,11 @@ export default class Spark {
     await assertOkResponse(res, 'Failed to fetch the current SPARK round')
     this.#activity.onHealthy()
-    const { retrievalTasks, ...round } = await res.json()
+    const { retrievalTasks, maxTaskPerNode, ...round } = await res.json()
```
is this a typo? We are mixing singular and plural `task` here

juliangruber replied:
https://github.com/filecoin-station/spark-api/pull/172/files has `maxTasksPerNode`. I think we should do some more testing here

bajtos replied:
Yeah, looks like a typo.

Do you have any suggestions on how to test this?

I think we also need to deploy https://github.com/filecoin-station/spark-api/pull/172 before we can test this new part.

juliangruber replied:
I think for now we would manually test this

juliangruber reviewed (changes requested):
Looks good, once `maxTasksPerRound` is fixed and this has been tested manually

bajtos commented:
I added another console log to give us visibility into the delay, and fixed the typo.

```
❯ zinnia run main.js
Getting current SPARK round details...
[10:25:17.699  INFO] SPARK started reporting retrievals
Current SPARK round: { roundId: "3876" }
  60 max tasks per node
  4000 retrieval tasks
(...)
Measurement submitted (id: 331077071)
[10:25:20.356 STATS] Jobs completed: 1
Sleeping for 57 seconds before starting the next task...
```

juliangruber commented:
+1
